src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -66,7 +66,7 @@ static std::string getDeviceFullName() {
 #else
 # error "Unkown CPU architecture. Please, add support to openvino/core/visibility.hpp"
 #endif
-    return brand_string;
+    return std::string(brand_string.c_str());
```

Comment:
Can you add test to catch issue in future.
I think this copy of string will just remove everything after first null termination character.

Maybe instead of coping to new string, remove all `\0` from existing and return same object.

```cpp
    // All `\0` will be at end the c_str should be displayed correctly but string size is same
    std::remove(brand_string.begin(), brand_string.end(), '\0');
```
or 
```cpp
    // Will remove all '\0`, string size will be reduce (some re-allocation can happen)
    brand_string.erase(std::remove(brand_string.begin(), brand_string.end(), '\0'), brand_string.end());
```

Reply:
I don't think `std::remove(brand_string.begin(), brand_string.end(), '\0');` can solve this issue because in cpu plugin `brand_string` capacity is fixed for x86 CPU, should be 3 * 4 * 4 = 48 bytes. After CPU info are filled, the left space will be kept by ` '\0' `, so `std::remove` will not work.

While `brand_string.erase(std::remove(brand_string.begin(), brand_string.end(), '\0'), brand_string.end());` can work, but as you said maybe memory allocation happen, it is not too much different with simple constructing `std::string(brand_string.c_str());`.

Another solution is to check non `\0` when copying data from `reg `to `brand_string`.

```
        for (size_t j = 0; j < sizeof(regs); j++)
            if (ch[j] != '\0')
                brand_string += ch[j];
```